src/user/PreferencesController.ts

```diff
@@ -188,7 +190,25 @@ export class PreferencesController extends BaseController<BaseConfig, Preference
       };
       return ids;
     }, {});
-    this.update({ identities: { ...identities } });
+
+    let { selectedAddress } = this.state;
+    if (!Object.keys(identities).includes(selectedAddress)) {
+      selectedAddress = Object.keys(identities)[0];
+    }
+    this.update({ identities: { ...identities }, selectedAddress });
+  }
+
+  /**
+   * Update the selected address.
+   *
+   * @param address - The new selected address
+   */
+  updateSelectedAddress(address: string) {
```

Comment:
This was added so that other controllers could update the selected address without using `update` directly, which would allow them to set the state to something invalid. This should be added in a separate PR.

Reply:
Oh, this method already exists :facepalm: It's below, `setSelectedAddress`. I'll remove this.